modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js

```diff
@@ -0,0 +1,204 @@
+		const propertyKey = firstContributorNotEmpty ? firstContributorNotEmpty.propertyKey : propertyGroups[0].propertyKey;
+
+		this.state = {
+			editingId: propertyKey
+		};
+	}
+
+	_handleCriteriaChange = (criteriaChange, index) => {
+		const {onQueryChange} = this.props;
+		onQueryChange(criteriaChange, index);
+	}
+
+	_handleCriteriaEdit = (id, editing) => {
+		this.setState(
+			{
+				editingId: editing ? undefined : id
+			}
+		);
+	}
+
+	_handlePreviewClick = () => {
+		this.props.onPreviewMembers();
+	}
+
+	_handleRootConjunctionClick = event => {
+		const {onConjunctionChange} = this.props;
+
+		onConjunctionChange();
+	}
+
+	render() {
+		const {
+			contributors,
+			editing,
+			emptyContributors,
+			membersCount,
+			membersCountLoading,
+			propertyGroups,
+			supportedConjunctions,
+			supportedOperators,
+			supportedPropertyTypes
+		} = this.props;
+
+		const {editingId} = this.state;
+
+		const rootClasses = getCN(
+			'contributor-builder-root',
+			{
+				editing
+			}
+		);
+
+		return (
+			<div className={rootClasses}>
+				<div className="criteria-builder-section-sidebar">
+					<CriteriaSidebar
+						onTitleClicked={this._handleCriteriaEdit}
+						propertyGroups={propertyGroups}
+						propertyKey={editingId}
+					/>
+				</div>
+
+				<div className="criteria-builder-section-main">
+					<div className="contributor-container">
+						<div className="container-fluid container-fluid-max-xl">
+							<div className="content-wrapper">
+								<div className="sheet">
+									<div className="d-flex flex-wrap justify-content-between mb-4">
+										<h2 className="sheet-title mb-2">{Liferay.Language.get('conditions')}</h2>
+										<div className="criterion-string">
+											<div className="btn-group">
+												<div className="btn-group-item inline-item">
+													<ClaySpinner
+														className="mr-4"
+														loading={membersCountLoading}
+														size="sm"
+													/>
+
+													{!membersCountLoading &&
+														<span className="mr-4">
+															{Liferay.Language.get('conditions-match')}
+															<b className="ml-2">
+																{getPluralMessage(
+																	Liferay.Language.get('x-member'),
+																	Liferay.Language.get('x-members'),
+																	membersCount
+																)}
+															</b>
+														</span>
+													}
+
+													<ClayButton
+														label={Liferay.Language.get('view-members')}
+														onClick={this._handlePreviewClick}
+														size="sm"
+														type="button"
+													/>
+												</div>
+											</div>
+										</div>
+									</div>
+
+									{emptyContributors && (editingId == undefined || !editing) &&
+										<EmptyPlaceholder />
+									}
+
+									{contributors.filter(
+										criteria => {
+											const editingCriteria = (editingId === criteria.propertyKey) && editing;
+											const emptyCriteriaQuery = criteria.query == '';
```

Comment:
We should aim to never use `==` and always use `===` instead. Double equals are much harder to debug.

Reply:
Fully agreed. Although, if don't mind I'm going to leave this one as it is because I think there are cases in which the backend provides `null` instead. But I'll try to get rid of it ASAP.

Reply:
Null shouldn't make a difference for this case.

`null == ''` is false
`null === ''` is false

Reply:
Thanks